### PR TITLE
Fix compile time explosion for minkowski distance

### PR DESCRIPTION
**Pull request by ahendriksen**
As explained in https://github.com/rapidsai/raft/issues/1246#issuecomment-1420828523, ptxas chokes on the minkowski distance when `VecLen==4` and `IdxT==uint32_t`.

This PR removes the veclen == 4 specialization for the minkowski distance.

Follow up to: #1239

### CODE REVIEW

**Review by cjnolet — CHANGES_REQUESTED** [review]
Not sure how I missed this one originally but I'd like to keep the (modern) naming conventions more consistent across the APIs. 

Also, I want to make sure the seemingly lower compile times aren't the result of sccache (which is suspect they are) and giving us false hope here. This is what happened before the release (we saw 45 minute compile times because of sccache, got excited, and then realized after the sccsche reset that the compile times were still taking >6 hours. I'd like to do some more profiling of this before merging.

File: cpp/include/raft/distance/masked_nn.cuh [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+ *                           start of group j + 1, i.e., it is the inclusive
+ *                           scan of the group lengths. The first group is
+ *                           always assumed to start at index 0 and the last
+ *                           group typically ends at index `n`. Length =
+ *                           `num_groups`.
+ * @param[out] out           will contain the reduced output (Length = `m`)
+ *                           (on device)
+ */
+template <typename DataT, typename OutT, typename IdxT, typename ReduceOpT, typename KVPReduceOpT>
+void maskedL2NN(raft::device_resources const& handle,
```
[the remark this answers is not in the document]
Comment (ahendriksen) [reply cpp/include/raft/distance/masked_nn.cuh:148]: Gladly!

**Comment by ahendriksen** [comment]
> Also, I want to make sure the seemingly lower compile times aren't the result of sccache

Did you reset sccache in commit https://github.com/rapidsai/raft/pull/1254/commits/585fc47d2183823d2bd21377dda35c615d52f00f or do we still have to test this?

**Comment by cjnolet** [comment]
I tried to but that didn't seem to work. I'm going to build locally without sccache in the meantime since my workstation specs are similar to CI.

**Review by cjnolet — APPROVED** [review]
I'm going ahead and approving this guy because I think the build issues related to these changes have been fixed. The other build issues predate these changes, I believe.

**Comment by cjnolet** [comment]
@ahendriksen just leaving a note for later that we should pull the `masked_l2_1nn_params` out from `masked_nn.cuh` and put it in a file called `mask_nn_types.hpp`. This allows users to expose that through their own APIs (knowing it doesn't require nvcc).

**Review by sean-frye — APPROVED** [review]
LGTM

**Pull request merged**